File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/history_summary.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class HistorySummary(object):
    """
    Summary information for a commit.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new HistorySummary object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param commit_id:
            The value to assign to the commit_id property of this HistorySummary.
        :type commit_id: str

        :param message:
            The value to assign to the message property of this HistorySummary.
        :type message: str

        :param author_name:
            The value to assign to the author_name property of this HistorySummary.
        :type author_name: str

        :param author_email:
            The value to assign to the author_email property of this HistorySummary.
        :type author_email: str

        :param time_created:
            The value to assign to the time_created property of this HistorySummary.
        :type time_created: datetime

        :param is_merge_commit:
            The value to assign to the is_merge_commit property of this HistorySummary.
        :type is_merge_commit: bool

        :param parents:
            The value to assign to the parents property of this HistorySummary.
        :type parents: list[str]

        """
        self.swagger_types = {
            'commit_id': 'str',
            'message': 'str',
            'author_name': 'str',
            'author_email': 'str',
            'time_created': 'datetime',
            'is_merge_commit': 'bool',
            'parents': 'list[str]'
        }

        self.attribute_map = {
            'commit_id': 'commitId',
            'message': 'message',
            'author_name': 'authorName',
            'author_email': 'authorEmail',
            'time_created': 'timeCreated',
            'is_merge_commit': 'isMergeCommit',
            'parents': 'parents'
        }

        self._commit_id = None
        self._message = None
        self._author_name = None
        self._author_email = None
        self._time_created = None
        self._is_merge_commit = None
        self._parents = None

    @property
    def commit_id(self):
        """
        **[Required]** Gets the commit_id of this HistorySummary.
        Commit SHA.


        :return: The commit_id of this HistorySummary.
        :rtype: str
        """
        return self._commit_id

    @commit_id.setter
    def commit_id(self, commit_id):
        """
        Sets the commit_id of this HistorySummary.
        Commit SHA.


        :param commit_id: The commit_id of this HistorySummary.
        :type: str
        """
        self._commit_id = commit_id

    @property
    def message(self):
        """
        **[Required]** Gets the message of this HistorySummary.
        First line of commit message (subject).


        :return: The message of this HistorySummary.
        :rtype: str
        """
        return self._message

    @message.setter
    def message(self, message):
        """
        Sets the message of this HistorySummary.
        First line of commit message (subject).


        :param message: The message of this HistorySummary.
        :type: str
        """
        self._message = message

    @property
    def author_name(self):
        """
        **[Required]** Gets the author_name of this HistorySummary.
        Commit author display name.


        :return: The author_name of this HistorySummary.
        :rtype: str
        """
        return self._author_name

    @author_name.setter
    def author_name(self, author_name):
        """
        Sets the author_name of this HistorySummary.
        Commit author display name.


        :param author_name: The author_name of this HistorySummary.
        :type: str
        """
        self._author_name = author_name

    @property
    def author_email(self):
        """
        Gets the author_email of this HistorySummary.
        Commit author email (optional).


        :return: The author_email of this HistorySummary.
        :rtype: str
        """
        return self._author_email

    @author_email.setter
    def author_email(self, author_email):
        """
        Sets the author_email of this HistorySummary.
        Commit author email (optional).


        :param author_email: The author_email of this HistorySummary.
        :type: str
        """
        self._author_email = author_email

    @property
    def time_created(self):
        """
        **[Required]** Gets the time_created of this HistorySummary.
        Commit authored time (or committed time if you prefer, but be consistent).


        :return: The time_created of this HistorySummary.
        :rtype: datetime
        """
        return self._time_created

    @time_created.setter
    def time_created(self, time_created):
        """
        Sets the time_created of this HistorySummary.
        Commit authored time (or committed time if you prefer, but be consistent).


        :param time_created: The time_created of this HistorySummary.
        :type: datetime
        """
        self._time_created = time_created

    @property
    def is_merge_commit(self):
        """
        Gets the is_merge_commit of this HistorySummary.
        True if commit has multiple parents.


        :return: The is_merge_commit of this HistorySummary.
        :rtype: bool
        """
        return self._is_merge_commit

    @is_merge_commit.setter
    def is_merge_commit(self, is_merge_commit):
        """
        Sets the is_merge_commit of this HistorySummary.
        True if commit has multiple parents.


        :param is_merge_commit: The is_merge_commit of this HistorySummary.
        :type: bool
        """
        self._is_merge_commit = is_merge_commit

    @property
    def parents(self):
        """
        Gets the parents of this HistorySummary.
        Parent commit SHAs (empty for root commit). Present to explain merge commits.


        :return: The parents of this HistorySummary.
        :rtype: list[str]
        """
        return self._parents

    @parents.setter
    def parents(self, parents):
        """
        Sets the parents of this HistorySummary.
        Parent commit SHAs (empty for root commit). Present to explain merge commits.


        :param parents: The parents of this HistorySummary.
        :type: list[str]
        """
        self._parents = parents

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/HistorySummary.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Summary information for a commit.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=HistorySummary.Builder.class)

public final class HistorySummary  {
    @Deprecated
    @java.beans.ConstructorProperties({"commitId", "message", "authorName", "authorEmail", "timeCreated", "isMergeCommit", "parents"})
    public HistorySummary(String commitId, String message, String authorName, String authorEmail, java.util.Date timeCreated, Boolean isMergeCommit, java.util.List<String> parents) {
        super();
        this.commitId = commitId;
        this.message = message;
        this.authorName = authorName;
        this.authorEmail = authorEmail;
        this.timeCreated = timeCreated;
        this.isMergeCommit = isMergeCommit;
        this.parents = parents;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Commit SHA.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("commitId")
private String commitId;

        /**
         * Commit SHA.
         * @param commitId the value to set
         * @return this builder
         **/
        

public Builder commitId(String commitId) {
    this.commitId = commitId;
    return this;
}
            /**
     * First line of commit message (subject).
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("message")
private String message;

        /**
         * First line of commit message (subject).
         * @param message the value to set
         * @return this builder
         **/
        

public Builder message(String message) {
    this.message = message;
    return this;
}
            /**
     * Commit author display name.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("authorName")
private String authorName;

        /**
         * Commit author display name.
         * @param authorName the value to set
         * @return this builder
         **/
        

public Builder authorName(String authorName) {
    this.authorName = authorName;
    return this;
}
            /**
     * Commit author email (optional).
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("authorEmail")
private String authorEmail;

        /**
         * Commit author email (optional).
         * @param authorEmail the value to set
         * @return this builder
         **/
        

public Builder authorEmail(String authorEmail) {
    this.authorEmail = authorEmail;
    return this;
}
            /**
     * Commit authored time (or committed time if you prefer, but be consistent).
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
private java.util.Date timeCreated;

        /**
         * Commit authored time (or committed time if you prefer, but be consistent).
         * @param timeCreated the value to set
         * @return this builder
         **/
        

public Builder timeCreated(java.util.Date timeCreated) {
    this.timeCreated = timeCreated;
    return this;
}
            /**
     * True if commit has multiple parents.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("isMergeCommit")
private Boolean isMergeCommit;

        /**
         * True if commit has multiple parents.
         * @param isMergeCommit the value to set
         * @return this builder
         **/
        

public Builder isMergeCommit(Boolean isMergeCommit) {
    this.isMergeCommit = isMergeCommit;
    return this;
}
            /**
     * Parent commit SHAs (empty for root commit). Present to explain merge commits.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("parents")
private java.util.List<String> parents;

        /**
         * Parent commit SHAs (empty for root commit). Present to explain merge commits.
         * @param parents the value to set
         * @return this builder
         **/
        

public Builder parents(java.util.List<String> parents) {
    this.parents = parents;
    return this;
}


        public HistorySummary build() {
            HistorySummary model = new HistorySummary(this.commitId
                , this.message
                , this.authorName
                , this.authorEmail
                , this.timeCreated
                , this.isMergeCommit
                , this.parents);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(HistorySummary model) {
                this.commitId(model.getCommitId());
    this.message(model.getMessage());
    this.authorName(model.getAuthorName());
    this.authorEmail(model.getAuthorEmail());
    this.timeCreated(model.getTimeCreated());
    this.isMergeCommit(model.getIsMergeCommit());
    this.parents(model.getParents());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Commit SHA.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("commitId")
    private final String commitId;

        /**
     * Commit SHA.
     * @return the value
     **/
    
    public String getCommitId() {
        return commitId;
    }


        /**
     * First line of commit message (subject).
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("message")
    private final String message;

        /**
     * First line of commit message (subject).
     * @return the value
     **/
    
    public String getMessage() {
        return message;
    }


        /**
     * Commit author display name.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("authorName")
    private final String authorName;

        /**
     * Commit author display name.
     * @return the value
     **/
    
    public String getAuthorName() {
        return authorName;
    }


        /**
     * Commit author email (optional).
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("authorEmail")
    private final String authorEmail;

        /**
     * Commit author email (optional).
     * @return the value
     **/
    
    public String getAuthorEmail() {
        return authorEmail;
    }


        /**
     * Commit authored time (or committed time if you prefer, but be consistent).
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
    private final java.util.Date timeCreated;

        /**
     * Commit authored time (or committed time if you prefer, but be consistent).
     * @return the value
     **/
    
    public java.util.Date getTimeCreated() {
        return timeCreated;
    }


        /**
     * True if commit has multiple parents.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isMergeCommit")
    private final Boolean isMergeCommit;

        /**
     * True if commit has multiple parents.
     * @return the value
     **/
    
    public Boolean getIsMergeCommit() {
        return isMergeCommit;
    }


        /**
     * Parent commit SHAs (empty for root commit). Present to explain merge commits.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("parents")
    private final java.util.List<String> parents;

        /**
     * Parent commit SHAs (empty for root commit). Present to explain merge commits.
     * @return the value
     **/
    
    public java.util.List<String> getParents() {
        return parents;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("HistorySummary(");
        sb.append("commitId=").append(String.valueOf(this.commitId));
        sb.append(", message=").append(String.valueOf(this.message));
        sb.append(", authorName=").append(String.valueOf(this.authorName));
        sb.append(", authorEmail=").append(String.valueOf(this.authorEmail));
        sb.append(", timeCreated=").append(String.valueOf(this.timeCreated));
        sb.append(", isMergeCommit=").append(String.valueOf(this.isMergeCommit));
        sb.append(", parents=").append(String.valueOf(this.parents));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof HistorySummary)) {
            return false;
        }

        HistorySummary other = (HistorySummary) o;
        return java.util.Objects.equals(this.commitId, other.commitId) &&
            java.util.Objects.equals(this.message, other.message) &&
            java.util.Objects.equals(this.authorName, other.authorName) &&
            java.util.Objects.equals(this.authorEmail, other.authorEmail) &&
            java.util.Objects.equals(this.timeCreated, other.timeCreated) &&
            java.util.Objects.equals(this.isMergeCommit, other.isMergeCommit) &&
            java.util.Objects.equals(this.parents, other.parents);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.commitId == null ? 43 : this.commitId.hashCode());
        result = (result * PRIME) + (this.message == null ? 43 : this.message.hashCode());
        result = (result * PRIME) + (this.authorName == null ? 43 : this.authorName.hashCode());
        result = (result * PRIME) + (this.authorEmail == null ? 43 : this.authorEmail.hashCode());
        result = (result * PRIME) + (this.timeCreated == null ? 43 : this.timeCreated.hashCode());
        result = (result * PRIME) + (this.isMergeCommit == null ? 43 : this.isMergeCommit.hashCode());
        result = (result * PRIME) + (this.parents == null ? 43 : this.parents.hashCode());
        return result;
    }


}
